src/main/java/seedu/condonery/logic/parser/ParserUtil.java

```diff
@@ -194,4 +197,16 @@ public static PropertyTypeEnum parsePropertyType(String propertyType) throws Par
                 "Invalid Property Type specified! Property Type must be one of HDB, CONDO, or LANDED");
         }
     }
+
+    /**
+     * Parses a string into {@code PropertyStatusEnum} enum
+     * @param propertyStatus string to PropertyStatusEnum
+     * @return a {@code PropertyStatusEnum} enum
+     * @throws ParseException if the string does not match any valid enum
+     */
+    public static PropertyStatusEnum parsePropertyStatus(String propertyStatus) throws ParseException {
```

Comment:
for this I think the exception is not being throw! because valueOf can throw IllegalArgumentException, need to explicitly check for it